homeassistant/components/ezviz/camera.py

```diff
@@ -1,197 +1,253 @@
-"""This component provides basic support for Ezviz IP cameras."""
+"""Support ezviz camera devices."""
 import asyncio
+from datetime import timedelta
 import logging
+from typing import Callable, List
 
-# pylint: disable=import-error
 from haffmpeg.tools import IMAGE_JPEG, ImageFrame
-from pyezviz.camera import EzvizCamera
-from pyezviz.client import EzvizClient, PyEzvizError
+from pyezviz.DeviceSwitchType import DeviceSwitchType
 import voluptuous as vol
 
 from homeassistant.components.camera import PLATFORM_SCHEMA, SUPPORT_STREAM, Camera
+from homeassistant.components.ffmpeg import DATA_FFMPEG
 from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
-from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import config_validation as cv, entity_platform
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ATTR_CAMERAS,
+    ATTR_DIRECTION,
+    ATTR_ENABLE,
+    ATTR_INFRARED_LIGHT,
+    ATTR_LEVEL,
+    ATTR_LIGHT,
+    ATTR_MOBILE_TRACKING,
+    ATTR_PRIVACY,
+    ATTR_SLEEP,
+    ATTR_SOUND,
+    ATTR_SPEED,
+    ATTR_SWITCH,
+    ATTR_TYPE,
+    CONF_FFMPEG_ARGUMENTS,
+    DATA_COORDINATOR,
+    DEFAULT_CAMERA_USERNAME,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_RTSP_PORT,
+    DIR_DOWN,
+    DIR_LEFT,
+    DIR_RIGHT,
+    DIR_UP,
+    DOMAIN,
+    MANUFACTURER,
+)
+from .coordinator import EzvizDataUpdateCoordinator
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_CAMERAS = "cameras"
-
-DEFAULT_CAMERA_USERNAME = "admin"
-DEFAULT_RTSP_PORT = "554"
-
-DATA_FFMPEG = "ffmpeg"
-
-EZVIZ_DATA = "ezviz"
-ENTITIES = "entities"
+MIN_TIME_BETWEEN_SESSION_RENEW = timedelta(seconds=90)
 
-CAMERA_SCHEMA = vol.Schema(
```

Comment:
Please keep the platform schema intact besides adding the deprecation validator.

Reply:
Done